compile/src/main/scala/sbt/compiler/RawCompiler.scala

```diff
@@ -21,14 +21,22 @@ class RawCompiler(val scalaInstance: xsbti.compile.ScalaInstance, cp: ClasspathO
 
     val arguments = compilerArguments(sources, classpath, Some(outputDirectory), options)
     log.debug("Plain interface to Scala compiler " + scalaInstance.actualVersion + "  with arguments: " + arguments.mkString("\n\t", "\n\t", ""))
-    val mainClass = Class.forName("scala.tools.nsc.Main", true, scalaInstance.loader)
-    val process = mainClass.getMethod("process", classOf[Array[String]])
-    process.invoke(null, arguments.toArray)
-    checkForFailure(mainClass, arguments.toArray)
+    val args = arguments.toArray
+    val reporter =
+      if (ScalaInstance.isDotty(scalaInstance.version)) {
+        val mainClass = Class.forName("dotty.tools.dotc.Main", true, scalaInstance.loader)
+        val process = mainClass.getMethod("process", classOf[Array[String]])
+        process.invoke(null, args)
+      } else {
+        val mainClass = Class.forName("scala.tools.nsc.Main", true, scalaInstance.loader)
+        val process = mainClass.getMethod("process", classOf[Array[String]])
+        process.invoke(null, arguments.toArray)
+        mainClass.getMethod("reporter").invoke(null)
+      }
+    checkForFailure(reporter, arguments.toArray)
   }
   def compilerArguments = new CompilerArguments(scalaInstance, cp)
-  protected def checkForFailure(mainClass: Class[_], args: Array[String]): Unit = {
-    val reporter = mainClass.getMethod("reporter").invoke(null)
+  protected def checkForFailure(reporter: AnyRef, args: Array[String]): Unit = {
```

Comment:
OK, will fix, I have no idea what is and isn't public API :). Is it okay if I push-force my branch or do you prefer code reviews to be addressed in separate commits?